pinot-core/src/main/java/org/apache/pinot/core/data/readers/ParquetRecordReader.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.readers;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.pinot.common.data.Schema;
+import org.apache.pinot.common.utils.ParquetUtils;
+import org.apache.pinot.core.data.GenericRow;
+import org.apache.pinot.core.util.AvroUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+
+
+/**
+ * Record reader for Parquet file.
+ */
+public class ParquetRecordReader implements RecordReader {
+  private final String _dataFilePath;
+  private final Schema _schema;
+
+  private Iterator<GenericRecord> _reader;
+
+  public ParquetRecordReader(File dataFile, Schema schema)
+      throws IOException {
+    _dataFilePath = "file://" + dataFile.getAbsolutePath();
```

Comment:
Is `file://` necessary?

Reply:
+1

Reply:
It is necessary to tell hadoop file system that this is a local file. 
I deleted it here and setting the default file URI schema to 'file://' in hadoop conf, which may looks more elegant now.